lib/data_classes/movie.dart

```diff
@@ -1,46 +1,88 @@
 class Movie {
+  final bool adult;
   final String backdrop;
-  final List<String> genres;
+  final List<int> genres;
+  final int id;
+  final String originalLanguage;
   final String originalTitle;
   final String overview;
+  final double popularity;
   final String poster;
   final DateTime releaseDate;
   final String title;
+  final bool video;
   final double voteAverage;
+  final int voteCount;
   int likes = 0;
 
   Movie(
-    this.title,
-    this.originalTitle,
+    this.adult,
+    this.backdrop,
     this.genres,
+    this.id,
+    this.originalLanguage,
+    this.originalTitle,
+    this.overview,
+    this.popularity,
     this.poster,
-    this.backdrop,
     this.releaseDate,
+    this.title,
+    this.video,
     this.voteAverage,
-    this.overview,
+    this.voteCount,
   );
 
+  factory Movie.fromJson(Map<String, dynamic> j) => Movie(
+        j['adult'],
+        'https://image.tmdb.org/t/p/w500${j['backdrop_path']}',
+        j['genre_ids'].cast<int>(),
+        j['id'],
+        j['original_language'],
+        j['original_title'],
+        j['overview'],
+        j['popularity'],
+        'https://image.tmdb.org/t/p/w500${j['poster_path']}',
```

Comment:
Let's parse the backdrop path as it is, and create a custom getter for the URL:
String get posterUrl => 'https://image.tmdb.org/t/p/w500${poster}';

Reply:
Done